File: qddm/qddm/DDNode.cpp
//  #===================================#
//  # Dynamic Data Node - Version 1.0.0 #
//  #                                   #
//  # Author: Marco Vagnoni             #
//  # Email:  marco.vagnoni@yahoo.com   #
//  # Date:   April     2019 (v1.0.0)   #
//  #===================================#

#include "DDNode.h"

DDNode::DDNode (int offset, void *object, int type)
{
    this->offset = offset;
    this->object = object;
    this->type   = type;
    next         = nullptr;
}

int     DDNode::getOffset () {return offset;}
DDNode *DDNode::getNext   () {return next;}
void   *DDNode::getObject () {return object;}
int     DDNode::getType   () {return type;}

void    DDNode::setNext   (DDNode *next) {this->next = next;}

File: qddm/jddm/src/main/java/it/xv2/ddm/DDNode.java
//  #===================================#
//  # Dynamic Data Node - Version 1.0.0 #
//  #                                   #
//  # Author: Marco Vagnoni             #
//  # Email:  marco.vagnoni@yahoo.com   #
//  # Date:   April     2019 (v1.0.0)   #
//  #===================================#

package it.xv2.ddm;

final class DDNode
{
    private DDNode       next;
    private final Object object;
    private final int    type;
    private final int    offset;

    DDNode (int offset, Object object, int type)
    {
        this.offset = offset;
        this.object = object;
        this.type   = type;
    }

    int    getOffset () {return offset;}
    DDNode getNext   () {return next;}
    Object getObject () {return object;}
    int    getType   () {return type;}

    void setNext (DDNode next) {this.next = next;}
}